src/wp-includes/cron.php

```diff
@@ -190,17 +238,32 @@ function wp_schedule_single_event( $timestamp, $hook, $args = array() ) {
  *                           hook's callback function. Each value in the array is passed
  *                           to the callback as an individual parameter. The array keys
  *                           are ignored. Default: empty array.
- * @return bool True if event successfully scheduled. False for failure.
+ * @param bool   $wp_error   Optional. Whether to return a WP_Error on failure. Default false.
+ * @return bool|WP_Error True if event successfully scheduled. False or WP_Error on failure.
  */
-function wp_schedule_event( $timestamp, $recurrence, $hook, $args = array() ) {
+function wp_schedule_event( $timestamp, $recurrence, $hook, $args = array(), $wp_error = false ) {
 	// Make sure timestamp is a positive integer.
 	if ( ! is_numeric( $timestamp ) || $timestamp <= 0 ) {
+		if ( $wp_error ) {
+			return new WP_Error(
+				'invalid_timestamp',
+				__( 'Event timestamp is not a positive integer' )
```

Comment:
This is a bit technical; "Event timestamp must be a valid Unix timestamp" or similar might be a bit better.